include/openthread/srp_client_service_pool.h

```diff
@@ -0,0 +1,175 @@
+/*
+ *  Copyright (c) 2021, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *  This file defines the OpenThread SRP (Service Registration Protocol) client item (service/host) pool.
+ */
+
+#ifndef OPENTHREAD_SRP_CLIENT_SERVICE_POOL_H_
+#define OPENTHREAD_SRP_CLIENT_SERVICE_POOL_H_
+
+#include <openthread/dns.h>
+#include <openthread/srp_client.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup api-srp
+ *
+ * @brief
+ *   This module includes functions for SRP client service pool.
+ *
+ * @{
+ *
+ * Functions in this module are only available when feature OPENTHREAD_CONFIG_SRP_CLIENT_SERVICE_POOL_ENABLE is enabled.
+ *
+ */
+
+/**
+ * This struct represents a SRP client service pool entry.
+ *
+ */
+typedef struct otSrpClientPoolServiceEntry
+{
+    otSrpClientService mService;  ///< The SRP client service structure
+    otDnsTxtEntry      mTxtEntry; ///< The SRP client TXT entry.
+} otSrpClientPoolServiceEntry;
+
+/**
+ * This function gets the string buffer to use for SRP client host name.
+ *
+ * @param[in] aInstance   A pointer to the OpenThread instance.
+ * @param[out] aSize      If not NULL, returns the size (number of bytes) of the string buffer.
+ *
+ * @returns A pointer to char buffer to use for SRP client host name.
+ *
+ */
+char *otSrpClientPoolGetHostNameString(otInstance *aInstance, uint16_t *aSize);
```

Comment:
The pool is really only for the service entries since we can have multiple services and add and remove them dynamically.

The host name and host address array mirror the SRP client APIs, where they are really single instances (e.g. if we us the API to set the host name, it is actually replacing any previous string).

Reply:
Should we rename to something like: `otSrpClientBufs*`?

Reply:
Renamed to use `otSrpClientBuffers*` in the last pushed commit.